project-koku/koku
# [COST-4328] Fix subs vCPU trino filter and ensure columns exist on aws_line_items table

## Jira Ticket

[COST-4328](https://issues.redhat.com/browse/COST-4328)

Will also address the missing column from https://issues.redhat.com/browse/COST-4199

## Description

This change will adjust the swatch sql to filter out empty strings since the `product_vcpu` column is a varchar and the null value is `''` instead of `null`. This will also add `product/physicalCores` and `product/vcpu` to the `TRINO_REQUIRED_COLUMNS` for AWS which will ensure those columns exist on the `aws_line_items` table. A trino migration to drop the `aws_line_items` table will follow this code out and once a new download occurs, the table will be recreated with the additional columns included. 

## Testing

1. Checkout Branch
2. Restart Koku
3. Load the test customer data.
4. Connect to trino and run a query:
5. `SELECT DISTINCT product_vcpu FROM aws_line_items;`
6. See there are a few values, such as:
```
trino:org1234567> SELECT DISTINCT product_vcpu FROM aws_line_items;
 product_vcpu 
--------------
 2            
              
 8.0          
 4.0          
 2.0          
(5 rows)
```
7. Run the same query using the existing filter: `SELECT DISTINCT product_vcpu FROM aws_line_items WHERE product_vcpu IS NOT NULL;`
8. See the same records, including the empty record we did not want:
```
trino:org1234567> SELECT DISTINCT product_vcpu FROM aws_line_items WHERE product_vcpu IS NOT NULL;
 product_vcpu 
--------------
 2            
              
 8.0          
 2.0          
 4.0          
(5 rows)
```
9. Change the where clause to the new one: `SELECT DISTINCT product_vcpu FROM aws_line_items WHERE product_vcpu != '';`
10. See that the empty record is now gone:
```
trino:org1234567> SELECT DISTINCT product_vcpu FROM aws_line_items WHERE product_vcpu != '';
 product_vcpu 
--------------
 2            
 4.0          
 8.0          
 2.0          
(4 rows)

Query 20231017_170107_02750_xq452, FINISHED, 1 node
http://localhost:8080/ui/query.html?20231017_170107_02750_xq452
Splits: 110 total, 110 done (100.00%)
CPU Time: 0.1s total,  338K rows/s, 53.4MB/s, 13% active
Per Node: 0.3 parallelism,  103K rows/s, 16.3MB/s
Parallelism: 0.3
Peak Memory: 3.66KB
0.25 [25.4K rows, 4.01MB] [103K rows/s, 16.3MB/s]
```
## Notes

...
